engine/access/rest/http/routes/collections.go

```diff
@@ -38,7 +38,7 @@ func GetCollectionByID(r *common.Request, backend access.API, link commonmodels.
 	var response commonmodels.Collection
 	err = response.Build(collection, transactions, link, r.ExpandFields)
 	if err != nil {
-		return nil, err
+		return nil, common.ErrorToStatusError(err)
```

Comment:
`common.ErrorToStatusError()` converts an `accessSentinel` error to a REST error. I don't think `response.Build()` calls into the backend code, so the returned error is not an `accessSentinel`. double check if the error returned is already wrapped. if not, you can use a specific status code error here.